modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/unified_query_builder/ScopeSelector.tsx

```diff
@@ -0,0 +1,225 @@
+import {openSelectionModal} from 'frontend-js-components-web';
+import React, {useContext, useMemo, useState} from 'react';
+
+import ThemeContext from '../../shared/ThemeContext';
+import {STATUS} from '../UQBEditSXPBlueprintForm';
+import CustomPanel from './shared/CustomPanel';
+
+type Status = STATUS.ACTIVE | STATUS.INACTIVE;
+
+type Scope = {
+	externalReferenceCode: string;
+	name: string;
+	status: Status;
+	type: string;
+};
+
+type Sorting = {
+	column: React.Key;
+	direction: 'ascending' | 'descending';
+};
+
+export default function ScopeSelector({
+	scope,
+	setScope,
+}: {
+	scope: Scope[];
+	setScope: (scope: Scope[]) => void;
+}) {
+	const [sort, setSort] = useState<Sorting>();
+
+	const {
+		namespace,
+		selectScopeURL,
+	}: {namespace: string; selectScopeURL: string} = useContext(ThemeContext);
+
+	const filteredScope = useMemo(() => {
+		if (!sort) {
+			return scope;
+		}
+
+		return scope.slice().sort((a, b) => {
+			let cmp = new Intl.Collator('en', {numeric: true}).compare(
+				a[sort.column as keyof Scope],
+				b[sort.column as keyof Scope]
+			);
+
+			if (sort.direction === 'descending') {
+				cmp *= -1;
+			}
+
+			return cmp;
+		});
+	}, [sort, scope]);
+
+	const _handleSelectScope = () => {
+		openSelectionModal({
+			id: `${namespace}selectScope`,
+			onSelect: (selectedItem: {
+				groupdescriptivename: string;
+				groupexternalreferencecode: string;
+				groupid: string;
+				groupscopelabel: string;
+			}) => {
+				if (!selectedItem) {
+					return;
+				}
+
+				if (
+					scope.find(
+						(item) =>
+							item.externalReferenceCode ===
+							selectedItem.groupexternalreferencecode
+					)
+				) {
+					return;
+				}
+
+				setScope([
+					...scope,
+					{
+						externalReferenceCode:
+							selectedItem.groupexternalreferencecode,
+						name: selectedItem.groupdescriptivename,
+						status: STATUS.ACTIVE,
+						type: selectedItem.groupscopelabel,
+					},
+				]);
+			},
+			selectEventName: `${namespace}selectScope`,
+			title: Liferay.Language.get('select-scope'),
+			url: selectScopeURL,
+		});
+	};
+
+	const _handleRemoveScope = (externalReferenceCode: string) => {
+		setScope(
+			scope.filter(
+				(item) => item.externalReferenceCode !== externalReferenceCode
+			)
+		);
+	};
+
+	return (
+		<CustomPanel
+			classNames="scope-selector"
+			title={Liferay.Language.get('scope')}
+		>
+			<>
+				<span className="text-4 text-secondary">
+					{Liferay.Language.get('scope-selector-description')}
+				</span>
+
+				<div className="c-mt-4">
+					<ClayButton
+						displayType="secondary"
+						onClick={_handleSelectScope}
+					>
+						<span className="inline-item inline-item-before">
+							<ClayIcon symbol="plus" />
+						</span>
+
+						{Liferay.Language.get('select-scope')}
+					</ClayButton>
+				</div>
+
+				{!!scope.length && (
+					<Table
+						columnsVisibility={false}
+						onSortChange={
+							setSort as (sorting: Sorting | null) => void
```

Comment:
I'd be cautious about using `as` because that's like you're overriding the type. If the ClayTable component changes its API in the future to return a different structure, TypeScript won't warn you because you forced the type with as.

Instead I think you'd want to change what the `state` expects:

```
const [sort, setSort] = useState<Sorting | null>();
```